src/ConfigServiceProvider.php

```diff
@@ -71,49 +71,32 @@ public function __construct(
         $prefix = self::DEFAULT_PREFIX,
         $separator = self::DEFAULT_SEPARATOR
     ) {
-        $this->config = [];
-
-        $config = ($config instanceof ApplicationConfig) ? $config : new ApplicationConfig($config, $separator);
+        $this->prefix = $prefix;
+        $this->config = ($config instanceof ApplicationConfig) ? $config : new ApplicationConfig($config, $separator);
+    }
 
+    public function boot()
+    {
         $configurator = new League\Configurator();
+        $configurator->addApplicationConfig($this->container, $this->config, $this->prefix);
 
-        $configurator->addApplicationConfig($config, $prefix);
-
-        if (isset($config[self::DEFAULT_DI_KEY])) {
-            $configurator->addServiceConfig(new ServiceConfig($config[self::DEFAULT_DI_KEY]));
+        if (isset($this->config[self::DEFAULT_DI_KEY])) {
+            $configurator->addServiceConfig(
+                $this->container,
+                new ServiceConfig($this->config[self::DEFAULT_DI_KEY])
+            );
         }
 
-
-        if (isset($config[self::DEFAULT_INFLECTORS_KEY])) {
-            $configurator->addInflectorConfig(new InflectorConfig($config[self::DEFAULT_INFLECTORS_KEY]));
+        if (isset($this->config[self::DEFAULT_INFLECTORS_KEY])) {
+            $configurator->addInflectorConfig(
+                $this->container,
+                new InflectorConfig($this->config[self::DEFAULT_INFLECTORS_KEY])
+            );
         }
-
-        $this->subProviders = $configurator->getServiceProvider();
-    }
-
-    public function provides($service = null)
-    {
-        return $this->subProviders->provides($service);
     }
 
     public function register()
     {
-        $this->subProviders->register();
-    }
-
-    public function boot()
-    {
-        $this->subProviders->boot();
-    }
-
-    public function setContainer(ContainerInterface $container)
```

Comment:
@tomphp 

Can we also remove the now unused import for `League\Container\ContainerInterface`?